tests/e2e/data_masking/test_data_masking.py

```diff
@@ -0,0 +1,129 @@
+import json
+from uuid import uuid4
+from aws_encryption_sdk.exceptions import DecryptKeyError
+import pytest
+from tests.e2e.utils import data_fetcher
+from aws_lambda_powertools.utilities.data_masking.base import DataMasking
+from aws_lambda_powertools.utilities.data_masking.providers.aws_encryption_sdk import AwsEncryptionSdkProvider
+
+
+@pytest.fixture
+def basic_handler_fn(infrastructure: dict) -> str:
+    return infrastructure.get("BasicHandler", "")
+
+
+@pytest.fixture
+def basic_handler_fn_arn(infrastructure: dict) -> str:
+    return infrastructure.get("BasicHandlerArn", "")
+
+
+@pytest.fixture
+def kms_key1_arn(infrastructure: dict) -> str:
+    return infrastructure.get("KMSKey1Arn", "")
+
+
+@pytest.fixture
+def kms_key2_arn(infrastructure: dict) -> str:
+    return infrastructure.get("KMSKey2Arn", "")
+
+
+@pytest.fixture
+def data_masker(kms_key1_arn) -> DataMasking:
+    return DataMasking(provider=AwsEncryptionSdkProvider(keys=[kms_key1_arn]))
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    # AWS Encryption SDK encrypt method only takes in bytes or strings
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting and then decrypting the encrypted data
+    encrypted_data = data_masker.encrypt(value)
+    decrypted_data = data_masker.decrypt(encrypted_data)
+
+    # THEN the result is the original input data
+    assert decrypted_data == value
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_context(data_masker):
+    # GIVEN an instantiation of DataMasking with the AWS encryption provider
+
+    value = bytes(str([1, 2, "string", 4.5]), "utf-8")
+
+    # WHEN encrypting and then decrypting the encrypted data with an encryption_context
+    encrypted_data = data_masker.encrypt(value, encryption_context={"this": "is_secure"})
+    decrypted_data = data_masker.decrypt(encrypted_data, encryption_context={"this": "is_secure"})
+
+    # THEN the result is the original input data
+    assert decrypted_data == value
+
+
+@pytest.mark.xdist_group(name="data_masking")
+def test_encryption_diff_context_fail(data_masker):
```

Comment:
```suggestion
def test_encryption_context_mismatch(data_masker):
```

nt: slightly clearer in the intent